### PR TITLE
Build and install ROCm via TheRock in AMD images

### PR DESCRIPTION
**Pull request by sogartar**

### CODE REVIEW

File: dockerfiles/amdgpu_ubuntu_jammy_ghr_x86_64.Dockerfile [hunk 1]
```diff
@@ -6,35 +6,30 @@ FROM docker.io/myoung34/github-runner:2.314.1-ubuntu-jammy
...
-    apt update && DEBIAN_FRONTEND=noninteractive apt install -y --no-install-recommends ca-certificates rocm-hip-libraries && \
-    echo "/opt/rocm/lib\n/opt/rocm/hip/lib" | tee /etc/ld.so.conf.d/24-rocm.conf && ldconfig && \
-    rm -rf /install-rocm/*
+# TheRock (ROCm)
+WORKDIR /install-the-rock
+COPY build_tools/install_the_rock.sh ./
+RUN ./install_the_rock.sh \
+  && rm -rf /install-the-rock
 WORKDIR /
-RUN rmdir /install-rocm/
```
Comment (sogartar) [inline dockerfiles/amdgpu_ubuntu_jammy_ghr_x86_64.Dockerfile:33]: I am curious why we don't do
```
RUN apt clean && rm -rf /var/lib/apt/lists/*
```
to remove the apt artifacts in the github image, but we do it in the ubuntu stock image? Is this intentional?

Comment (stellaraccident) [reply dockerfiles/amdgpu_ubuntu_jammy_ghr_x86_64.Dockerfile:33]: I'd have to look at each, but generally adding that as a separate run step doesn't help anything. Each RUN gets exported as a layer, so doing it separately just means that you have all of that in a previous layer and then you make a final later that records the deletions (but doesn't actually save any image space).

The right way to do package stuff is to script it all in one script then, in I've RUN invoke the script and && any cleanup tasks. 

Not all docker images were done right.

**Review by stellaraccident — APPROVED** [review]
May want to come back and fuse the cleanup steps.

File: dockerfiles/amdgpu_ubuntu_jammy_ghr_x86_64.Dockerfile (hunk 1 above)
Comment (sogartar) [reply dockerfiles/amdgpu_ubuntu_jammy_ghr_x86_64.Dockerfile:33]: I added the cleanup to the github image as well.

Comment (sogartar) [reply dockerfiles/amdgpu_ubuntu_jammy_ghr_x86_64.Dockerfile:33]: The problem is that multiple steps may use apt and then if you cleanup after each step then the next step would repeat stuff and slow down the image creation.

**Pull request merged**